box-share-sdk/src/main/java/com/box/androidsdk/share/sharerepo/ShareRepo.java

```
@@ -105,4 +120,106 @@ public void inviteCollabs(BoxCollaborationItem boxCollaborationItem, BoxCollabor
     public LiveData<BoxResponse<BoxItem>> getItemInfo() {
         return mItemInfo;
     }
+
+    /**
+     * Create a default shared link of an item.
+     * @param boxCollaborationItem the item to get share link on
+     */
+    public void createDefaultSharedLink(BoxCollaborationItem boxCollaborationItem) {
+        handleTaskAndPostValue(mController.createDefaultSharedLink(boxCollaborationItem), mSharedLinkedItem);
+    }
+
+    /**
+     * Disable shared link of an item.
+     * @param boxCollaborationItem the item to disable share link on
+     */
+    public void disableSharedLink(BoxCollaborationItem boxCollaborationItem) {
+        handleTaskAndPostValue(mController.disableShareLink(boxCollaborationItem), mSharedLinkedItem);
+    }
+
+    /**
+     * Change the download permission of a share item
+     * @param boxCollaborationItem the item to change download permission on
+     * @param canDownload the new download permission
+     */
+    public void changeDownloadPermission(BoxCollaborationItem boxCollaborationItem, boolean canDownload) {
+        if (boxCollaborationItem instanceof BoxFile) {
+            handleTaskAndPostValue(mController.executeRequest(BoxItem.class, ((BoxRequestsFile.UpdatedSharedFile) mController.getCreatedSharedLinkRequest(boxCollaborationItem)).setCanDownload(canDownload)), mSharedLinkedItem);
+        }
+        else if (boxCollaborationItem instanceof BoxFolder) {
+            handleTaskAndPostValue(mController.executeRequest(BoxItem.class, ((BoxRequestsFolder.UpdateSharedFolder) mController.getCreatedSharedLinkRequest(boxCollaborationItem)).setCanDownload(canDownload)), mSharedLinkedItem);
+        } else {
+            //an invalid request
```

better to throw an exception instead of silently swallowing it.